docs/evaluation/convolution.rst

```diff
@@ -25,32 +25,175 @@ or the redistribution of the counts as a function of energy
+
+One-dimensional convolution with a PSF
+======================================
+
+In this example we want to convolve a model by the "kernel"
+[5, 10, 3, 2], which requires creating a
+:py:class:`sherpa.data.Data1D` instance to store the
+data, and then pass it to :py:class:`sherpa.instrument.PSFModel`:
+
+    >>> from sherpa.data import Data1D
+    >>> from sherpa.instrument import PSFModel
+    >>> k = np.asarray([5, 10, 3, 2])
+    >>> x = np.arange(k.size)
+    >>> kernel = Data1D('kdata1', x, k)
+    >>> psf1 = PSFModel('psf1', kernel=kernel)
+    >>> print(psf1)
+    psf1
+       Param        Type          Value          Min          Max      Units
+       -----        ----          -----          ---          ---      -----
+       psf1.kernel  frozen       kdata1
+       psf1.radial  frozen            0            0            1
+       psf1.norm    frozen            1            0            1
+
+To show how this model works we will use it to convolve
+a :py:class:`~sherpa.models.basic.Box1D` model (this is chosen
+since it has sharp edges and so the convolution is more obvious).
+
+    >>> from sherpa.models.basic import Box1D
+    >>> point = Box1D('pt')
+    >>> point.xlow = 9.5
+    >>> point.xhi = 10.5
+    >>> print(point)
+    point
+       Param        Type          Value          Min          Max      Units
+       -----        ----          -----          ---          ---      -----
+       point.xlow   thawed          9.5 -3.40282e+38  3.40282e+38
+       point.xhi    thawed         10.5 -3.40282e+38  3.40282e+38
+       point.ampl   thawed            1           -1            1
+
+The convolution case is created by applying the `psf1` model
+to the `point` model (the :ref:`2D example <convolution-psf2d-convolve>`
+below shows an example of applying a kernel to a
+composite model):
+
+    >>> convolved = psf1(pt)
+    >>> print(convolved)
+    psfmodel(point)
+       Param        Type          Value          Min          Max      Units
+       -----        ----          -----          ---          ---      -----
+       point.xlow   thawed          9.5 -3.40282e+38  3.40282e+38
+       point.xhi    thawed         10.5 -3.40282e+38  3.40282e+38
+       point.ampl   thawed            1           -1            1
+
+.. _convolution-1d-fold:
+
+The model can be evaluated bith before and after convolution, by
+passing it the data grid. Unlike normal model evaluation the
+`PSFModel` class requires that its
+:py:meth:`~sherpa.instrument.PSFModel.fold` model be called before
+evaluation. This method pre-calculates terms needed for the
+convolution (which is done using a fourier transform), and so needs
+the grid over which it is to be applied. This is done by passing in a
+:py:class:`~sherpa.data.Data` instance (in this case the Y data in the
+:py:class:`~sherpa.data.Data1D` instance is not used so is set to
+zero):
+
+    >>> x = np.arange(6, 15)
+    >>> blank = Data1D('blank', x, np.zeros(x.size))
+    >>> psf1.fold(blank)
```

Comment:
This strikes me as complex. Why not do that when the model is called wit the x-values of whatever data comes in. The result can then be cached so that the fold is called only once. We have caching already in Sherpa, so that can be re-used, alternatively a simple
```
self.folded_x = None
def call (...):
    if self.folded_x != x_value_passed_in:
       self.fold(...)
```
will do. I realize that that's not part of this PR, but I wonder if we can fix that instead of spending the work documenting a complicated procedure.

Reply:
I am not a fan of the current `fold` interface (I think it should either be more explicit or handled differently). However, I have no time or energy to work on this, and I do not believe we have the resources to work on this before the end of the year. So I really would like to get some documentation out there to users.

Reply:
Makes sense.